.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -2697,7 +2697,7 @@ stages:
 
 - stage: trace_pipeline
   condition: eq(variables['isScheduledBuild'], 'False')
-  dependsOn: [integration_tests_windows, integration_tests_windows_iis, msi_integration_tests_windows, integration_tests_linux, unit_tests_linux, unit_tests_macos, unit_tests_windows, unit_tests_arm64, master_commit_id, integration_tests_arm64, benchmarks, throughput, throughput_appsec, tool_artifacts_tests_windows, tool_artifacts_tests_linux, upload_to_s3, upload_to_azure, installer_smoke_tests, installer_smoke_tests_arm64]
+  dependsOn: [integration_tests_windows, integration_tests_windows_iis, msi_integration_tests_windows, integration_tests_linux, unit_tests_linux, unit_tests_macos, unit_tests_windows, unit_tests_arm64, master_commit_id, integration_tests_arm64, benchmarks, throughput, throughput_appsec, tool_artifacts_tests_windows, tool_artifacts_tests_linux, upload_to_s3, upload_to_azure, installer_smoke_tests, installer_smoke_tests_arm64, coverage]
```

Comment:
I take it the "just put it at the end" approach didn't work 🙁

Reply:
Yes, when you do that, it just depends on the last stage defined in the yaml, not all of them. So it could have worked if I had moved the coverage in last but one, but I figured we would forget it next time we add a step, whereas with dependencies we can still forget it but I thought the chances of forgetting were slightly reduced as we use dependencies everywhere